### ISSUE TITLE
@pika/plugin-ts-standard-pkg gives lots of errors about relative imports

### ISSUE THREAD

**Issue by icopp** [pikapkg/builders#3]
I gave a quick try using my package [basedirs](https://github.com/icopp/basedirs), using this configuration:

```
  "@pika/pack": {
    "pipeline": [
      ["@pika/plugin-ts-standard-pkg", { "exclude": ["__tests__/*","*.test.*"] }],
      ["@pika/plugin-build-node"],
      ["@pika/plugin-build-web"],
      ["@pika/plugin-bundle-node"],
      ["@pika/plugin-bundle-web"],
      ["@pika/plugin-simple-bin", {"bin": "basedirs" }]
    ]
  }
```

The build seems to work, but gives a whole boatload of errors:

```
dist-src/lib/__tests__/cachedir.test.js
  ⚠️  [3:0] "../cachedir": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/__tests__/configdir.test.js
  ⚠️  [3:0] "../configdir": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/__tests__/configdirs.test.js
  ⚠️  [3:0] "../configdirs": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/__tests__/datadir.test.js
  ⚠️  [3:0] "../datadir": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/__tests__/datadirs.test.js
  ⚠️  [3:0] "../datadirs": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/__tests__/datalocaldir.test.js
  ⚠️  [3:0] "../datalocaldir": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/__tests__/executabledir.test.js
  ⚠️  [3:0] "../executabledir": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/__tests__/homedir.test.js
  ⚠️  [2:0] "../homedir": Valid relative imports must include the ".js" file extension.
  ⚠️  [3:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/__tests__/runtimedir.test.js
  ⚠️  [3:0] "../runtimedir": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "../constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/cachedir.js
  ⚠️  [3:0] "./constants": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "./util/get-windows-known-folder-path": Valid relative imports must include the ".js" file extension.
  ⚠️  [5:0] "./util/to-sanitized-name": Valid relative imports must include the ".js" file extension.
  ⚠️  [6:0] "./util/to-snake-case-name": Valid relative imports must include the ".js" file extension.
dist-src/lib/configdir.js
  ⚠️  [3:0] "./constants": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "./util/get-windows-known-folder-path": Valid relative imports must include the ".js" file extension.
  ⚠️  [5:0] "./util/to-sanitized-name": Valid relative imports must include the ".js" file extension.
  ⚠️  [6:0] "./util/to-snake-case-name": Valid relative imports must include the ".js" file extension.
dist-src/lib/configdirs.js
  ⚠️  [3:0] "./constants": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "./util/get-windows-known-folder-path": Valid relative imports must include the ".js" file extension.
dist-src/lib/datadir.js
  ⚠️  [3:0] "./constants": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "./util/get-windows-known-folder-path": Valid relative imports must include the ".js" file extension.
  ⚠️  [5:0] "./util/to-snake-case-name": Valid relative imports must include the ".js" file extension.
  ⚠️  [6:0] "./util/to-sanitized-name": Valid relative imports must include the ".js" file extension.
dist-src/lib/datadirs.js
  ⚠️  [3:0] "./constants": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "./util/get-windows-known-folder-path": Valid relative imports must include the ".js" file extension.
dist-src/lib/datalocaldir.js
  ⚠️  [3:0] "./constants": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "./util/get-windows-known-folder-path": Valid relative imports must include the ".js" file extension.
  ⚠️  [5:0] "./datadir": Valid relative imports must include the ".js" file extension.
  ⚠️  [6:0] "./util/to-sanitized-name": Valid relative imports must include the ".js" file extension.
dist-src/lib/executabledir.js
  ⚠️  [3:0] "./constants": Valid relative imports must include the ".js" file extension.
  ⚠️  [4:0] "./util/get-windows-known-folder-path": Valid relative imports must include the ".js" file extension.
dist-src/lib/runtimedir.js
  ⚠️  [2:0] "./constants": Valid relative imports must include the ".js" file extension.
dist-src/lib/util/get-windows-known-folder-path.js
  ⚠️  [2:0] "../constants": Valid relative imports must include the ".js" file extension.
  ⚠️  [3:0] "./parse-windows-guid": Valid relative imports must include the ".js" file extension.
dist-src/lib/util/to-snake-case-name.js
  ⚠️  [1:0] "./to-sanitized-name": Valid relative imports must include the ".js" file extension.
```

**Comment by FredKSchott** [comment]
Yea, I hate this too. This is caused by this missing feature: https://github.com/Microsoft/TypeScript/issues/16577

**tl;dr:**
- This builders goal is to build a standard ES2018 `dist-src/`
- imports without file extensions aren't valid in ES2018, that something that Node.js introduced and bundlers/transpilers/TS followed.
- We believe the developer should be able to skip the file extension if they prefer to, especially in TS where it's a bit confusing to import a relative TypeScript file with the `.js` extension. `@pika/plugin-standard-pkg` automatically resolves missing file extensions for you, for example.
- Unfortunately `tsc` doesn't support an option for this yet

I'm open to suggestions about short-term solutions, until TypeScript supports this.

**Comment by pgilad** [comment]
Since the Typescript team might, or might not, add this, and this project will automatically follow future updates, the end user currently has nothing much to do, so I suggest that in the meantime this warning should be muted (or add a configuration to mute it).

If there's no valid action I can take, don't show me the warning :)

**Comment by JoshRosenstein** [comment]
👍 

> Since the Typescript team might, or might not, add this, and this project will automatically follow future updates, the end user currently has nothing much to do, so I suggest that in the meantime this warning should be muted (or add a configuration to mute it).
> 
> If there's no valid action I can take, don't show me the warning :)

Agreed. or having a flag to disable different settings within the [babel-validate-file](https://unpkg.com/standard-pkg@0.4.0/dist-src/util/babel-validate-file.js) depending on the build. Or isnt there a babel plugin to transform these imports after tsc compiles and run before linting?

Currently I got `465 standard-pkg errors found`- all relative imports errors, so it kinda makes the report useless if i cant determine if there are any other errors that arent associated with `relative imports`

**Comment by FredKSchott** [comment]
This still doesn't appear to be supported by typescript core, but this may be the most promising direction for us to take in the meantime: https://github.com/Microsoft/TypeScript/issues/16577#issuecomment-448540664

If we could build our own transform plugin for TS, then we could use https://github.com/cevek/ttypescript/ to run that in our ts build plugin. Is anyone interested in taking a look at this?

**Comment by FredKSchott** [comment]
I'm not sure if TS is ever going to prioritize this, so I've added an option to standard-pkg for TS projects to ignore file extensions in relative imports. Pika packages are moving more towards per-package bundling anyway, so this is becoming less of an issue for us and I know that this output is frustrating for a lot of people. 

Expect these warnings to go around in the next batch of plugin updates (for TS projects only)

**Comment by FredKSchott** [comment]
Fixed by 1942e761b595cd35a4755abc6e98eb3e15bd11a3, will go out with the next release
